File: jetpack_java/src/main/java/com/kunminx/jetpack_java/sample_04_databinding/ui/adapter/DataBindingLocationAdapter.java


package com.kunminx.jetpack_java.sample_04_databinding.ui.adapter;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.kunminx.jetpack_java.R;
import com.kunminx.jetpack_java.common_data.bean.LocationBean;
import com.kunminx.jetpack_java.common_ui.adapter.DiffUtilCallbacks;
import com.kunminx.jetpack_java.databinding.AdapterLocationDatabindingBinding;
import com.kunminx.architecture.ui.adapter.SimpleBindingAdapter;

/**
 * Create by KunMinX at 2020/5/31
 */
public class DataBindingLocationAdapter extends SimpleBindingAdapter<LocationBean, AdapterLocationDatabindingBinding> {

    public DataBindingLocationAdapter(Context context) {
        super(context, R.layout.adapter_location_databinding, new DiffUtilCallbacks().getLocationBeanItemCallback());
    }

    @Override
    protected void onBindItem(AdapterLocationDatabindingBinding binding, LocationBean item, RecyclerView.ViewHolder holder) {
        binding.setBean(item);
    }
}

File: jetpack_kotlin/src/main/java/com/example/jetpack_kotlin/sample_04_databinding/adapter/DataBindingLocationAdapter.kt
package com.example.jetpack_kotlin.sample_04_databinding.adapter

import android.content.Context
import androidx.recyclerview.widget.RecyclerView
import com.example.jetpack_koltin.R
import com.example.jetpack_koltin.databinding.AdapterLocationDatabindingBinding
import com.example.jetpack_kotlin.common_data.bean.LocationBean
import com.example.jetpack_kotlin.common_ui.adapter.locationDiffCallback
import com.kunminx.architecture.ui.adapter.SimpleBindingAdapter

/**
 * @author Flywith24
 * @date   2020/5/31
 * time   22:56
 * description
 */
class DataBindingLocationAdapter(context: Context) : SimpleBindingAdapter<LocationBean, AdapterLocationDatabindingBinding>(context, R.layout.adapter_location_databinding, locationDiffCallback) {
    override fun onBindItem(binding: AdapterLocationDatabindingBinding?, item: LocationBean?, holder: RecyclerView.ViewHolder?) {
        binding?.bean = item
    }
}